authui/src/resendButton.ts

```diff
@@ -1,40 +1,45 @@
-export function setupResendButton(): () => void {
-  const el = document.querySelector("#resend-button") as HTMLButtonElement;
-  if (el == null) {
-    return () => {};
-  }
+import { Controller } from "@hotwired/stimulus";
 
-  const scheduledAt = new Date();
-  const cooldown = Number(el.getAttribute("data-cooldown")) * 1000;
-  const label = el.getAttribute("data-label");
-  const labelUnit = el.getAttribute("data-label-unit")!;
-  let animHandle: number | null = null;
+export class ResendButtonController extends Controller {
+  static targets = ["button"];
 
-  function tick() {
-    const now = new Date();
-    const timeElapsed = now.getTime() - scheduledAt.getTime();
+  declare buttonTarget: HTMLButtonElement;
+  declare animHandle: number | null;
 
-    let displaySeconds = 0;
-    if (timeElapsed <= cooldown) {
-      displaySeconds = Math.round((cooldown - timeElapsed) / 1000);
-    }
+  connect() {
+    const button = this.buttonTarget;
 
-    if (displaySeconds === 0) {
-      el.disabled = false;
-      el.textContent = label;
-      animHandle = null;
-    } else {
-      el.disabled = true;
-      el.textContent = labelUnit.replace("%d", String(displaySeconds));
-      animHandle = requestAnimationFrame(tick);
-    }
-  }
+    const scheduledAt = new Date();
+    const cooldown = Number(button.getAttribute("data-cooldown")) * 1000;
+    const label = button.getAttribute("data-label");
+    const labelUnit = button.getAttribute("data-label-unit")!;
+
+    const tick = () => {
+      const now = new Date();
+      const timeElapsed = now.getTime() - scheduledAt.getTime();
 
-  animHandle = requestAnimationFrame(tick);
+      let displaySeconds = 0;
+      if (timeElapsed <= cooldown) {
+        displaySeconds = Math.round((cooldown - timeElapsed) / 1000);
+      }
 
-  return () => {
-    if (animHandle != null) {
-      cancelAnimationFrame(animHandle);
+      if (displaySeconds === 0) {
+        button.disabled = false;
+        button.textContent = label;
+        this.animHandle = null;
+      } else {
+        button.disabled = true;
+        button.textContent = labelUnit.replace("%d", String(displaySeconds));
+        this.animHandle = requestAnimationFrame(tick);
+      }
+    };
+
+    this.animHandle = requestAnimationFrame(tick);
+  }
+
+  disconnect() {
+    if (this.animHandle != null) {
+      cancelAnimationFrame(this.animHandle);
```

Comment:
We should set `this.animHandle` to `null` after cancelling the animation frame. Generally speaking, a cleanup method may be called more than once. The cleanup method should be able to be run more than once, and running it the second time should have no effect. Here, the same animation handle will be freed twice and it is a problem.

Reply:
Done, please take a look